networkx/algorithms/tests/test_dag.py

```diff
@@ -635,29 +645,14 @@ def test_is_aperiodic_bipartite():
     assert not nx.is_aperiodic(G)
 
 
-def test_is_aperiodic_rary_tree():
-    G = nx.full_rary_tree(3, 27, create_using=nx.DiGraph())
-    assert not nx.is_aperiodic(G)
-
-
-def test_is_aperiodic_disconnected():
-    # disconnected graph
+def test_is_aperiodic_single_node():
     G = nx.DiGraph()
-    nx.add_cycle(G, [1, 2, 3, 4])
-    nx.add_cycle(G, [5, 6, 7, 8])
+    G.add_node(0)
     assert not nx.is_aperiodic(G)
-    G.add_edge(1, 3)
-    G.add_edge(5, 7)
+    G.add_edge(0, 0)
     assert nx.is_aperiodic(G)
 
 
-def test_is_aperiodic_disconnected2():
```

Comment:
Can we leave a test of disconnected? It can test for the raised exception due to being disconnected. And I guess we could test both a completely disconnected graph and a weakly but not strongly connected graph.

Reply:
We already have a test for weakly but not strongly connected graph `test_is_aperiodic_not_strongly_connected_raises`, so I will simply restore a disconnected test and no need for a new weakly connected test.

Now we have two tests `test_is_aperiodic_weakly_connected_raises` and `test_is_aperiodic_disconnected_raises`.